lib/libutee/tee_api_operations.c

```diff
@@ -414,13 +437,28 @@ TEE_Result TEE_SetOperationKey2(TEE_OperationHandle operation,
 	operation->info.handleState &= ~TEE_HANDLE_FLAG_KEY_SET;
 
 	if (key1 != TEE_HANDLE_NULL) {
-		TEE_CopyObjectAttributes(operation->key1, key1);
-		TEE_CopyObjectAttributes(operation->key2, key2);
+		res = TEE_CopyObjectAttributes1(operation->key1, key1);
+		if (res != TEE_SUCCESS)
+			goto err;
+
+		res = TEE_CopyObjectAttributes1(operation->key2, key2);
```

Comment:
If this function call returns TEE_ERROR_CORRUPT_OBJECT, we should return TEE_ERROR_CORRUPT_OBJECT_2 instead.